patch/field_type.go

```diff
@@ -0,0 +1,201 @@
+package patch
+
+import (
+	"go/ast"
+	"go/types"
+	"log"
+	"strings"
+
+	"golang.org/x/tools/go/ast/astutil"
+)
+
+func (p *Patcher) patchTypeDef(id *ast.Ident, obj types.Object) {
+	fieldType, ok := p.fieldTypes[obj]
+	if !ok {
+		return
+	}
+
+	parent := p.findParentNode(id)
+	if pkg, name, isSlice := packageAndName(fieldType); pkg != "" {
+		f := p.fileOf(id)
+		pkgImport := packageImport(pkg)
+		astutil.AddNamedImport(p.fset, f, pkgImport, pkg)
+		fieldType = pkgImport + "." + name
+		if isSlice {
+			fieldType = "[]" + fieldType
+		}
+	}
+	castDecl := func(v *ast.Field) bool {
+		switch t := v.Type.(type) {
+		case *ast.Ident:
+			t.Name = fieldType
+			return true
+		case *ast.ArrayType:
+			if isSliceType(fieldType) {
```

Comment:
Is `isSliceType` needed if a user can’t specify `[]T` in this PR?

Reply:
Same as the previous comment. But this case could maybe be removed.

Reply:
I think it makes sense to pare this down to the minimum amount of code to make it work as documented.

Reply:
I'll remove the part that is no more required of this case statement.